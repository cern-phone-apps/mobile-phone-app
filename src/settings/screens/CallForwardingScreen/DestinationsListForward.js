import React from 'react';
import { ScrollView} from 'react-native';
import { ListItem } from 'react-native-elements';
import { IconButton, RadioButton } from 'react-native-paper';
import PropTypes from 'prop-types';
import ColorPalette from '../../../styles/ColorPalette';

function DestinationsListForward(props) {
  const {
    localForwardList,
    removeLocalForwardNumber,
    enabledForwardNumber,
    setEnabledForwardNumber
  } = props;

  const onChangeForward = newValue => {
    console.log(`Onchange forward to ${newValue}`);
    setEnabledForwardNumber(newValue);
  };

  return (
    <ScrollView style={{ flex: 1 }}>
      <RadioButton.Group
        onValueChange={onChangeForward}
        value={enabledForwardNumber}
      >
        {localForwardList.map(number => (
          <ListItem
            key={number.text}
            title={number.value}
            bottomDivider
            leftElement={() => (
              <RadioButton color={ColorPalette.primary} value={number.value} />
            )}
            rightElement={() => (
              <IconButton
                testID={`icon-button-${number.value}`}
                color="#000"
                icon="delete"
                onPress={() => {
                  if (enabledForwardNumber === number.value) {
                    setEnabledForwardNumber('');
                  }
                  removeLocalForwardNumber(number);
                }}
              />
            )}
          />
        ))}
      </RadioButton.Group>
    </ScrollView>
  );
}

DestinationsListForward.propTypes = {
  localForwardList: PropTypes.arrayOf(PropTypes.string),
  enabledForwardNumber: PropTypes.string,
  setEnabledForwardNumber: PropTypes.func.isRequired,
  removeLocalForwardNumber: PropTypes.func.isRequired
};

DestinationsListForward.defaultProps = {
  localForwardList: [],
  enabledForwardNumber: ''
};
export default DestinationsListForward;
