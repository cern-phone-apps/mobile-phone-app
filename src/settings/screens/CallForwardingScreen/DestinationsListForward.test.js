import React from 'react';
import {
  waitForElement,
  render,
  fireEvent
} from '@testing-library/react-native';
import DestinationsListForward from './DestinationsListForward';

const removeLocalForwardNumber = jest.fn(() => {});
const setEnabledForwardNumber = jest.fn(() => {});

test('Render the component and press save', async () => {
  const { baseElement, queryByTestId } = render(
    <DestinationsListForward
      removeLocalForwardNumber={removeLocalForwardNumber}
      enabledForwardNumber="64444"
      setEnabledForwardNumber={setEnabledForwardNumber}
      localForwardList={[
        {
          text: 'one',
          value: '64443'
        },
        {
          text: 'two',
          value: '64444'
        },
        {
          text: 'three',
          value: '64445'
        }
      ]}
    />
  );

  expect(baseElement).toMatchSnapshot();
  expect(queryByTestId('icon-button-64443')).not.toBeNull();
  expect(queryByTestId('icon-button-64444')).not.toBeNull();
  expect(queryByTestId('icon-button-64445')).not.toBeNull();
  const button = queryByTestId('icon-button-64444');
  fireEvent.press(button);
  expect(setEnabledForwardNumber).toHaveBeenCalledTimes(1);
  expect(removeLocalForwardNumber).toHaveBeenCalledTimes(1);
  const button2 = queryByTestId('icon-button-64443');
  fireEvent.press(button2);
  expect(setEnabledForwardNumber).toHaveBeenCalledTimes(1);
  expect(removeLocalForwardNumber).toHaveBeenCalledTimes(2);
});
