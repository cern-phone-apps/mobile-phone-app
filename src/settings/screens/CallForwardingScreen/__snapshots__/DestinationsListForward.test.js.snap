// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the component and press save 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <TouchableHighlight
            accessibilityComponentType="radiobutton_unchecked"
            accessibilityLiveRegion="polite"
            accessibilityRole="button"
            accessibilityTraits="button"
            activeOpacity={0.85}
            color="#22529e"
            delayPressOut={100}
            disabled={false}
            status="unchecked"
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
            underlayColor="rgba(34, 82, 158, 0.33999999999999997)"
            value="64443"
          >
            <View
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#22529e",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </TouchableHighlight>
          <View
            width={16}
          />
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              64443
            </Text>
          </View>
          <View
            width={16}
          />
          <TouchableHighlight
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityTraits="button"
            activeOpacity={0.85}
            centered={true}
            delayPressOut={100}
            disabled={false}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 18,
                    "height": 36,
                    "justifyContent": "center",
                    "margin": 6,
                    "overflow": "visible",
                    "width": 36,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
            testID="icon-button-64443"
            underlayColor="rgba(0, 0, 0, 0.16)"
          >
            <View>
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </TouchableHighlight>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <TouchableHighlight
            accessibilityComponentType="radiobutton_checked"
            accessibilityLiveRegion="polite"
            accessibilityRole="button"
            accessibilityTraits="button"
            activeOpacity={0.85}
            color="#22529e"
            delayPressOut={100}
            disabled={false}
            status="checked"
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
            underlayColor="rgba(34, 82, 158, 0.33999999999999997)"
            value="64444"
          >
            <View
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#22529e",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </TouchableHighlight>
          <View
            width={16}
          />
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              64444
            </Text>
          </View>
          <View
            width={16}
          />
          <TouchableHighlight
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityTraits="button"
            activeOpacity={0.85}
            centered={true}
            delayPressOut={100}
            disabled={false}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 18,
                    "height": 36,
                    "justifyContent": "center",
                    "margin": 6,
                    "overflow": "visible",
                    "width": 36,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
            testID="icon-button-64444"
            underlayColor="rgba(0, 0, 0, 0.16)"
          >
            <View>
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </TouchableHighlight>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <TouchableHighlight
            accessibilityComponentType="radiobutton_unchecked"
            accessibilityLiveRegion="polite"
            accessibilityRole="button"
            accessibilityTraits="button"
            activeOpacity={0.85}
            color="#22529e"
            delayPressOut={100}
            disabled={false}
            status="unchecked"
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
            underlayColor="rgba(34, 82, 158, 0.33999999999999997)"
            value="64445"
          >
            <View
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#22529e",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </TouchableHighlight>
          <View
            width={16}
          />
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              64445
            </Text>
          </View>
          <View
            width={16}
          />
          <TouchableHighlight
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityTraits="button"
            activeOpacity={0.85}
            centered={true}
            delayPressOut={100}
            disabled={false}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 18,
                    "height": 36,
                    "justifyContent": "center",
                    "margin": 6,
                    "overflow": "visible",
                    "width": 36,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
            testID="icon-button-64445"
            underlayColor="rgba(0, 0, 0, 0.16)"
          >
            <View>
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </TouchableHighlight>
        </View>
      </View>
    </ScrollView>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": "100%",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingBottom": 4,
          "paddingEnd": 4,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <TouchableWithoutFeedback
        hitSlop={
          Object {
            "bottom": 4,
            "left": 4,
            "right": 4,
            "top": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(250, 186, 48, 0.95)",
              },
              Object {
                "borderRadius": 14,
                "height": 28,
                "justifyContent": "center",
                "paddingHorizontal": 12,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 12,
                "includeFontPadding": false,
                "lineHeight": 16,
              }
            }
          >
            Dismiss All
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
    <ScrollView
      data={
        Array [
          Object {
            "category": "Warning: Failed prop type: Invalid prop \`localForwardList[0]\` of type \`object\` supplied to \`DestinationsListForward\`, expected \`string\`.
    in DestinationsListForward",
            "warnings": Array [
              YellowBoxWarning {
                "message": Object {
                  "content": "Warning: Failed prop type: Invalid prop \`localForwardList[0]\` of type \`object\` supplied to \`DestinationsListForward\`, expected \`string\`.
    in DestinationsListForward",
                  "substitutions": Array [],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 15,
                    "file": "/Users/fiser/developer/mobile-phone-app/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 20,
                    "methodName": "printWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 11,
                    "file": "/Users/fiser/developer/mobile-phone-app/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 82,
                    "methodName": "checkPropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/fiser/developer/mobile-phone-app/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1796,
                    "methodName": "validatePropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/fiser/developer/mobile-phone-app/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1969,
                    "methodName": "Object.createElementWithValidation [as createElement]",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/fiser/developer/mobile-phone-app/src/settings/screens/CallForwardingScreen/DestinationsListForward.test.js",
                    "lineNumber": 14,
                    "methodName": "_callee$",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      scrollEnabled={false}
      scrollEventThrottle={50}
      scrollsToTop={false}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "height": 48.5,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(250, 186, 48, 0.95)",
                },
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "flex": 1,
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                <Text>
                  Warning: Failed prop type: Invalid prop \`localForwardList[0]\` of type \`object\` supplied to \`DestinationsListForward\`, expected \`string\`.
    in DestinationsListForward
                </Text>
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </ScrollView>
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "rgba(250, 186, 48, 0.95)",
          "marginTop": 0.5,
        }
      }
    />
  </View>
</View>
`;
